tracer/src/Datadog.Trace/Agent/Transports/GzipCompressedContent.cs

```diff
@@ -0,0 +1,52 @@
+// <copyright file="GzipCompressedContent.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+#if NETCOREAPP
+using System.IO;
+using System.IO.Compression;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Agent.Transports;
+
+internal class GzipCompressedContent : HttpContent
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<GzipCompressedContent>();
+
+    private readonly HttpContent _content;
+
+    public GzipCompressedContent(HttpContent content)
+    {
+        // Copy original headers
+        foreach (var header in content.Headers)
+        {
+            Headers.TryAddWithoutValidation(header.Key, header.Value);
+        }
+
+        Headers.ContentEncoding.Add("gzip");
+        _content = content;
+    }
+
+    protected override async Task SerializeToStreamAsync(Stream stream, TransportContext? context)
+    {
+        Log.Debug("GZip compressing payload...");
+#pragma warning disable CA2007
+        await using var gzip = new GZipStream(stream, CompressionMode.Compress, true);
+#pragma warning restore CA2007
+        await _content.CopyToAsync(gzip).ConfigureAwait(false);
```

Comment:
`IAsyncDisposable` isn't available in .NET Core 2.1, which will mean this breaks when we switch from netstandard2.0.

Of course, other stuff will break too, but up to now we've chosen to use the normal `IDisposable` everywhere, and use an explicit flush instead to avoid any risk of sync over async
```suggestion
        using var gzip = new GZipStream(stream, CompressionMode.Compress, true);
        await _content.CopyToAsync(gzip).ConfigureAwait(false);
        await gzip.FlushAsync().ConfigureAwait(false);
```

bonus: it gets rid of the `#pragma` 😅

Reply:
done in [790fc27](https://github.com/DataDog/dd-trace-dotnet/pull/5060/commits/790fc27dd9297aedb21e3091d87043f8e05e23f2)